test/integration/connect/envoy/helpers.bash

```diff
@@ -966,3 +966,57 @@ function create_peering {
   # echo "$output" >&3
   [ "$status" == 0 ]
 }
+
+function get_lambda_envoy_http_filter {
+  local HOSTPORT=$1
+  local NAME_PREFIX=$2
+  run retry_default curl -s -f $HOSTPORT/config_dump
+  [ "$status" -eq 0 ]
+  # get the full http filter object so the individual fields can be validated.
+  echo "$output" | jq --raw-output ".configs[2].dynamic_listeners[] | .active_state.listener.filter_chains[].filters[] | select(.name == \"envoy.filters.network.http_connection_manager\") | .typed_config.http_filters[] | select(.name == \"envoy.filters.http.aws_lambda\") | .typed_config"
+}
+
+function register_lambdas {
+  local DC=${1:-primary}
+  # register lambdas to the catalog
+  for f in $(find workdir/${DC}/register -type f -name 'lambda_*.json'); do
+    retry_default curl -sL -XPUT -d @${f} "http://localhost:8500/v1/catalog/register" >/dev/null && \
+      echo "Registered Lambda: $(jq -r .Service.Service $f)"
+  done
+}
+
+function assert_lambda_envoy_dynamic_cluster_exists {
+  local HOSTPORT=$1
+  local NAME_PREFIX=$2
+  local REGION=$3
+  local PORT=$4
+
+  local BODY=$(get_envoy_dynamic_cluster_once $HOSTPORT $NAME_PREFIX)
+  [ -n "$BODY" ]
+
+  [ "$(echo $BODY | jq -r '.cluster.type')" == "LOGICAL_DNS" ]
+  [ "$(echo $BODY | jq -r '.cluster.transport_socket.typed_config.sni')" == '*.amazonaws.com' ]
+  [ "$(echo $BODY | jq -r '.cluster.dns_lookup_family')" == "V4_ONLY" ]
+  [ "$(echo $BODY | jq -r '.cluster.metadata.filter_metadata|.["com.amazonaws.lambda"].egress_gateway')" == "true" ]
```

Comment:
I added more validation for the lambda filters than there seems to be for the other envoy config.. The config fields are important but if this is too verbose I can thin this out.

Reply:
I think we can safely remove most of these cases. At this level of testing, the most important thing is that Lambda is configured at all. We have a bunch of golden tests that capture the details ([snapshot](https://github.com/hashicorp/consul/blob/main/agent/xds/testdata/serverless_plugin/listeners/lambda-terminating-gateway-with-service-resolvers.latest.golden#L153-L160), [test](https://github.com/hashicorp/consul/blob/main/agent/xds/serverless_plugin_oss_test.go)).

Reply:
Done. I kept the check to make sure the function ARN is correct, but removed the rest.